### DISCUSSION TITLE
The picture become hazy after using vips_resize

### DISCUSSION THREAD

**Discussion by zhifengzhuang** [libvips/libvips#2389] [Q&A]
Here is my picture.
![F9031320-8C25-4EF6-B4B8-5A3E7A1E4F9A](https://user-images.githubusercontent.com/13492988/128868707-1d263bf2-ca7a-48b7-b16a-d39ff54bae59.png)
I resize the picture using commands bellow.
```
# vips-8.11.2-Sat Jul  3 14:17:10 UTC 2021
vips resize F9031320-8C25-4EF6-B4B8-5A3E7A1E4F9A.png out.png 0.8
```
![image](https://user-images.githubusercontent.com/13492988/128869081-0970137c-c095-47c1-9922-258a0b0d04fd.png)
As you can see, the round edge I point out seems to become hazy and black.
Is there any way to solve this?

**Comment by jcupitt** [comment]
Hello @781545872,

You need to premultiply transparent images before resize. Try this:

```
$ vips premultiply circle.png x.v 
$ vips resize x.v x2.v 0.8
$ vips unpremultiply x2.v small.png
```

To make:

![x2](https://user-images.githubusercontent.com/580843/129007640-546fd1d4-9ab7-437b-a25d-63905b356941.png)

The `thumbnail` operation will do this for you automatically.

**Reply by zhifengzhuang** [reply]
![image](https://user-images.githubusercontent.com/13492988/129006737-f3d52ee2-94a4-476e-981e-2058a559eec2.png)
Is it normal to see a gray line at what I point out?
The origin one seems to be more glossy.

**Reply by jcupitt** [reply]
I updated my answer. Try refreshing the page.

**Reply by zhifengzhuang** [reply]
Thanks! It works. @jcupitt
